jwst/outlier_detection/utils.py

```diff
@@ -215,51 +215,49 @@ def median_with_resampling(input_models,
         A WCS corresponding to the median data.
 
     median_error : np.ndarray, optional
-        A median error estimate, returned only if `return_error` is True.
+        A median error estimate, returned only if `return_error` is `True`
+        *and* ``resamp.compute_err`` is "driz_err".
 
     """
-    if not resamp.single:
-        raise ValueError("median_with_resampling should only be used for resample_many_to_many")
-
     in_memory = not input_models._on_disk
     indices_by_group = list(input_models.group_indices.values())
     ngroups = len(indices_by_group)
 
+    return_error = return_error and (resamp.compute_err == "driz_err")
```

Comment:
I actually would like to discuss this with you. In particular, we could still return `median_err` just have it `None` or a `numpy` array of `NaN`s. Or we could issue a log message and not return a `median_err`. The problem with this approach is that the calling code still expects 3 return values but could (unexpectedly) get just two. What would you prefer?

Reply:
I think no matter what it would be nice to have a log message.  With regards to returns, IMO it would be clearest if median_err were set to None; that way (as you said) there would be no need to handle a variable number of outputs but it's blatantly obvious that it was not computed.

Downstream code could more easily check ``if median_err is None`` than ``if np.all(np.isnan(median_err))``, and downstream code expecting a valid ``median_err`` would be more likely to fail instead of giving unexpected/bad results, which I view as a positive.